messaging/src/main/java/org/axonframework/queryhandling/DefaultQueryBusSpanFactory.java

```diff
@@ -0,0 +1,173 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.queryhandling;
+
+import org.axonframework.common.BuilderUtils;
+import org.axonframework.tracing.Span;
+import org.axonframework.tracing.SpanFactory;
+
+/**
+ * Default implementation of the {@link QueryBusSpanFactory}. Can be configured to include the query handling of a
+ * distributed query in the same trace or not (true by default).
+ *
+ * @author Mitchell Herrijgers
+ * @since 4.9.0
+ */
+public class DefaultQueryBusSpanFactory implements QueryBusSpanFactory {
+
+    private final SpanFactory spanFactory;
+    private final boolean distributedInSameTrace;
+
+    /**
+     * Creates a new {@link DefaultQueryBusSpanFactory} using the provided {@code builder}.
+     *
+     * @param builder The builder to build the {@link DefaultQueryBusSpanFactory} from.
+     */
+    protected DefaultQueryBusSpanFactory(Builder builder) {
+        builder.validate();
+        this.spanFactory = builder.builderSpanFactory;
+        this.distributedInSameTrace = builder.distributedInSameTrace;
+    }
+
+
+    @Override
+    public Span createQuerySpan(QueryMessage<?, ?> queryMessage, boolean distributed) {
+        if (distributed) {
+            return spanFactory.createDispatchSpan(() -> "QueryBus.queryDistributed", queryMessage);
+        }
+        return spanFactory.createInternalSpan(() -> "QueryBus.query", queryMessage);
+    }
+
+    @Override
+    public Span createSubscriptionQuerySpan(SubscriptionQueryMessage<?, ?, ?> queryMessage, boolean distributed) {
+        if (distributed) {
+            return spanFactory.createDispatchSpan(() -> "QueryBus.subscriptionQueryDistributed", queryMessage);
+        }
+        return spanFactory.createInternalSpan(() -> "QueryBus.subscriptionQuery", queryMessage);
+    }
+
+    @Override
+    public Span createSubscriptionQueryProcessUpdateSpan(SubscriptionQueryUpdateMessage<?> updateMessage,
+                                                         SubscriptionQueryMessage<?, ?, ?> queryMessage) {
+        return spanFactory.createChildHandlerSpan(() -> "QueryBus.subscriptionQuery", updateMessage, queryMessage);
+    }
+
+    @Override
+    public Span createScatterGatherSpan(QueryMessage<?, ?> queryMessage, boolean distributed) {
+        if (distributed) {
+            return spanFactory.createDispatchSpan(() -> "QueryBus.scatterGatherQueryDistributed", queryMessage);
+        }
+        return spanFactory.createInternalSpan(() -> "QueryBus.scatterGatherQuery", queryMessage);
+    }
+
+    @Override
+    public Span createScatterGatherHandlerSpan(QueryMessage<?, ?> queryMessage, int handlerIndex) {
+        return spanFactory.createInternalSpan(() -> "QueryBus.scatterGatherQuery-" + handlerIndex, queryMessage);
```

Comment:
Wouldn't there still be value in naming the response handling differently from the query dispatching as we did in the previous version?

Reply:
Sure, we can keep that name